protocol/synchronizer/synchronizer.go

```diff
@@ -338,12 +362,15 @@ func (s *Synchronizer) AdvanceView(syncInfo hotstuff.SyncInfo) {
 	s.startTimeoutTimer()
 
 	s.logger.Debugf("advanced to view %d", newView)
-	s.eventLoop.AddEvent(ViewChangeEvent{View: newView, Timeout: timeout})
+	s.eventLoop.AddEvent(hotstuff.ViewChangeEvent{View: newView, Timeout: timeout})
 
 	leader := s.leaderRotation.GetLeader(newView)
 	if leader == s.opts.ID() {
-		s.consensus.Propose(syncInfo)
-	} else if replica, ok := s.configuration.Replica(leader); ok {
+		sInfo, ok := s.consensus.Propose(s.View(), s.highQC, syncInfo)
```

Comment:
I think the return value here can actually be omitted. 
Passing back the same QC will not do anything new. 
The important part is to receive the QC when OnPropose is called from the network. See comments in consensus.go.